README


This archive contains the distribution Dist-Metadata,
version 0.925:

  Information about a perl module distribution

This software is copyright (c) 2011 by Randy Stauner.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Dist-Metadata

0.925     2013-02-08T02:40:09Z

  - Specify 1.03 as minimum required version of Digest
    to ensure that it tries Digest::SHA.  Closes gh-6.

0.924     2012-11-10T17:38:33Z

  - Add 'module_info' method for getting additional data (like checksums)
    with the provided module hashref.  Thanks to Jeffrey Ryan Thalhammer
    for the suggestion.
  - Add 'file_checksum' method to the Dist::Metadata::Dist base class.

0.923     2012-06-19T03:13:55Z

  - By default, when determining packages, filter out packages
    that don't match the file name (ones that can't be loaded via use/require).
    This mimics the behavior of PAUSE.
    If you want the old behavior of including "inner" packages in the meta
    "provides" set "include_inner_packages" to true in the constructor.
    Thanks to Jeffrey Ryan Thalhammer for finding and fixing this
    (and including tests)!

0.922     2011-10-19T22:30:53Z

  - Let CPAN::Meta set release_status instead of defaulting to 'stable'.
    Fixes an issue with dev/alpha versions.
    Thanks to Jeffrey Ryan Thalhammer (thaljef) for reporting and fixing.

0.921     2011-09-09T23:10:21Z

  - Fix 'no_index' checking with non-unix paths.

0.920     2011-08-18T04:11:45Z

  [API Change]
  - Dist::extract_into() and Dist::physical_directory() now return just $dir
    in scalar context (and continue to return ($dir, @files) in list context)

  [Enhancements]
  - Support zip archives (in addition to tar files).
    (Thanks to thaljef for reporting.)

  [Fixes]
  - Always include inc/, t/, and xt/ in meta/no_index.
    (Thanks to thaljef for reporting.)
  - Make return value of Dir::physical_directory() consistent with
    other subclasses.

0.915     2011-08-16T18:52:01Z

  - Remove warning for no packages found (thaljef).

0.914     2011-08-02T16:01:46Z

  - Use Path::Class::Dir's mkpath for simplicity/compatibility.
    Closes RT 69961.  Thanks to MELEZHIK for reporting.

0.913     2011-07-19T21:15:24Z

  - Pass paths to CPAN::DistnameInfo in Unix format.
    Thanks to jeroenl for figuring out that test failure on Win32.
    Closes RT 69585.

  - Remove unnecessary build file from release tarball

0.912     2011-07-14T23:38:40Z

  - Use CPAN::DistnameInfo to parse name/version from file name

0.911     2011-07-04T22:10:24Z

  - Fix test to use non-os specific paths in regexp

0.910     2011-07-03T00:06:00Z

  [Fixes]
  - Work around weird, possible perl bug (described at http://bit.ly/mhaQ4x)
    that made Struct->file_content return undef for IO-like objects

  [Prereqs]
  - use Path::Class instead of File::Spec
  - ensure File::Spec::Native is available

  [Documentation]
  - Clarify details of module's purpose in Pod Description

  [Testing]
  - Add dist format tests
  - Report version of JSON::PP (if any) installed to help debug
    troublesome smoke testing environments

0.904     2011-06-24T06:09:23Z

  - Fix path mismatch for "provides" when using Module::Metadata on Win32
  - Add more TODO ideas

0.903     2011-06-23T05:59:41Z

  - Load File::Spec (and subclasses) consistently

0.902     2011-06-23T05:51:42Z

  [Fixes]
  - Change determine_packages to always return paths in unix format
    (as required by CPAN::Meta::Spec)
  - Fix regexp path bugs on windows

  [Enhancements]
  - Include 'xt/' in the default no_index:directory list

  [Documentation]
  - Add items to pod TODO list

0.901     2011-06-22T01:22:07Z

  - Initial release


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "File::Spec" => 0,
    "Module::Build" => "0.3601",
    "Test::Fatal" => 0,
    "Test::MockObject" => "1.09",
    "Test::More" => "0.96"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Information about a perl module distribution",
  "dist_author" => [
    "Randy Stauner <rwstauner\@cpan.org>"
  ],
  "dist_name" => "Dist-Metadata",
  "dist_version" => "0.921",
  "license" => "perl",
  "module_name" => "Dist::Metadata",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Archive::Tar" => 1,
    "Archive::Zip" => "1.30",
    "CPAN::DistnameInfo" => "0.12",
    "CPAN::Meta" => "2.1",
    "Carp" => 0,
    "File::Basename" => 0,
    "File::Find" => 0,
    "File::Spec::Native" => "1.002",
    "File::Temp" => "0.19",
    "List::Util" => 0,
    "Module::Metadata" => 0,
    "Path::Class" => "0.24",
    "Try::Tiny" => "0.09",
    "constant" => 0,
    "parent" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


corpus/structs.pl
#
# This file is part of Dist-Metadata
#
# This software is copyright (c) 2011 by Randy Stauner.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
$Dist::Metadata::Test::Structs = {
  'nometafile_dev_release' => {
    'Dist-Metadata-Test-NoMetaFile-DevRelease-0.1_1/README' => 'This "dist" is for testing Dist::Metadata.
',
    'Dist-Metadata-Test-NoMetaFile-DevRelease-0.1_1/lib/Dist/Metadata/Test/NoMetaFile/DevRelease.pm' => 'package Dist::Metadata::Test::NoMetaFile::DevRelease;
# ABSTRACT: Fake dist for testing metadata determination

our $VERSION = \'0.1_1\';
'
  },
  'index_like_pause' => {
    'Dist-Metadata-Test-LikePause-0.1/README' => 'This "dist" is for testing Dist::Metadata.
',
    'Dist-Metadata-Test-LikePause-0.1/lib/Dist/Metadata/Test/LikePause.pm' => 'package Dist::Metadata::Test::LikePause;

# ABSTRACT: Fake dist for testing metadata determination

our $VERSION = \'0.1\';

# This should be excluded unless "include_inner_packages" is true
package ExtraPackage;

our $VERSION = \'0.2\';
'
  },
  'metafile' => {
    'Dist-Metadata-Test-MetaFile-2.2/lib/Dist/Metadata/Test/MetaFile.pm' => 'package Dist::Metadata::Test::MetaFile;
# ABSTRACT: Fake dist for testing metadata determination

# does not match META file but we trust the META file
our $VERSION = \'1.5\';
',
    'Dist-Metadata-Test-MetaFile-2.2/lib/Dist/Metadata/Test/MetaFile/PM.pm' => 'package Dist::Metadata::Test::MetaFile::PM;
# ABSTRACT: Just a file to be indexed

our $VERSION = \'1.1\';
',
    'Dist-Metadata-Test-MetaFile-2.2/META.yml' => '---
abstract: Fake dist for testing metadata determination
author:
- Randy Stauner <rwstauner@cpan.org>
dynamic_config: 0
generated_by: hand
license:
- perl_5
meta-spec:
  url: http://search.cpan.org/perldoc?CPAN::Meta::Spec
  version: \'2\'
name: Dist-Metadata-Test-MetaFile
no_index:
  directory:
  - corpus
  - examples
  - inc
  - share
  - t
  - xt
provides:
  Dist::Metadata::Test::MetaFile:
    file: lib/Dist/Metadata/Test/MetaFile.pm
    version: \'2.05\'
  Dist::Metadata::Test::MetaFile::PM:
    file: lib/Dist/Metadata/Test/MetaFile/PM.pm
    version: \'2.04\'
release_status: stable
version: \'2.2\'
',
    'Dist-Metadata-Test-MetaFile-2.2/META.json' => '{
   "abstract" : "Fake dist for testing metadata determination",
   "author" : [
      "Randy Stauner <rwstauner@cpan.org>"
   ],
   "dynamic_config" : 0,
   "generated_by" : "hand",
   "license" : [
      "perl_5"
   ],
   "meta-spec" : {
      "url" : "http://search.cpan.org/perldoc?CPAN::Meta::Spec",
      "version" : "2"
   },
   "name" : "Dist-Metadata-Test-MetaFile",
   "no_index" : {
      "directory" : [
         "corpus",
         "examples",
         "inc",
         "share",
         "t",
         "xt"
      ]
   },
   "provides" : {
      "Dist::Metadata::Test::MetaFile" : {
         "file" : "lib/Dist/Metadata/Test/MetaFile.pm",
         "version" : "2.1"
      },
      "Dist::Metadata::Test::MetaFile::PM" : {
         "file" : "lib/Dist/Metadata/Test/MetaFile/PM.pm",
         "version" : "2.0"
      }
   },
   "release_status" : "stable",
   "version" : "2.2"
}

',
    'Dist-Metadata-Test-MetaFile-2.2/README' => 'This "dist" is for testing Dist::Metadata.
'
  },
  'metafile_incomplete' => {
    'Dist-Metadata-Test-MetaFile-Incomplete-2.1/lib/Dist/Metadata/Test/MetaFile/Incomplete.pm' => 'package Dist::Metadata::Test::MetaFile::Incomplete;
# ABSTRACT: Just a file to be indexed

our $VERSION = \'2.1\';
',
    'Dist-Metadata-Test-MetaFile-Incomplete-2.1/README' => 'This "dist" is for testing Dist::Metadata.
',
    'Dist-Metadata-Test-MetaFile-Incomplete-2.1/META.yml' => '---
abstract: Fake dist for testing metadata determination
author:
- Randy Stauner <rwstauner@cpan.org>
dynamic_config: 0
generated_by: hand
license:
- perl_5
meta-spec:
  url: http://search.cpan.org/perldoc?CPAN::Meta::Spec
  version: \'2\'
name: Dist-Metadata-Test-MetaFile-Incomplete
no_index:
  directory:
  - examples
  - share
  - xt
provides: {}
release_status: stable
version: \'2.2\'
',
    'Dist-Metadata-Test-MetaFile-Incomplete-2.1/inc/NotThis.pm' => 'package NotThis;
# ABSTRACT: Not to be indexed
1;
',
    'Dist-Metadata-Test-MetaFile-Incomplete-2.1/t/lib/Never.pm' => 'package Never;
# ABSTRACT: Never index this

1;
',
    'Dist-Metadata-Test-MetaFile-Incomplete-2.1/META.json' => '{
   "abstract" : "Fake dist for testing metadata determination",
   "author" : [
      "Randy Stauner <rwstauner@cpan.org>"
   ],
   "dynamic_config" : 0,
   "generated_by" : "hand",
   "license" : [
      "perl_5"
   ],
   "meta-spec" : {
      "url" : "http://search.cpan.org/perldoc?CPAN::Meta::Spec",
      "version" : "2"
   },
   "name" : "Dist-Metadata-Test-MetaFile-Incomplete",
   "no_index" : {
      "directory" : [
         "examples",
         "share",
         "xt"
      ]
   },
   "provides" : {},
   "release_status" : "stable",
   "version" : "2.1"
}

'
  },
  'nometafile' => {
    'Dist-Metadata-Test-NoMetaFile-0.1/README' => 'This "dist" is for testing Dist::Metadata.
',
    'Dist-Metadata-Test-NoMetaFile-0.1/lib/Dist/Metadata/Test/NoMetaFile/PM.pm' => 'package Dist::Metadata::Test::NoMetaFile::PM;
# ABSTRACT: Just a file to be indexed

our $VERSION = \'0.1\';
',
    'Dist-Metadata-Test-NoMetaFile-0.1/lib/Dist/Metadata/Test/NoMetaFile.pm' => 'package Dist::Metadata::Test::NoMetaFile;
# ABSTRACT: Fake dist for testing metadata determination

our $VERSION = \'0.1\';
'
  },
  'noroot' => {
    'lib/Dist/Metadata/Test/NoRoot.pm' => 'package Dist::Metadata::Test::NoRoot;
# ABSTRACT: Fake dist for testing metadata determination

our $VERSION = \'3.3\';
',
    'lib/Dist/Metadata/Test/NoRoot/PM.pm' => 'package Dist::Metadata::Test::NoRoot::PM;
# ABSTRACT: Just a file to be indexed

our $VERSION = \'3.25\';
',
    'README' => 'This "dist" is for testing Dist::Metadata.
'
  },
  'subdir' => {
    'Dist-Metadata-Test-SubDir-1.5/lib/Dist/Metadata/Test/SubDir.pm' => 'package Dist::Metadata::Test::SubDir;
# ABSTRACT: Fake dist for testing metadata determination

our $VERSION = \'1.1\';
',
    'Dist-Metadata-Test-SubDir-1.5/lib/Dist/Metadata/Test/SubDir/PM.pm' => 'package Dist::Metadata::Test::SubDir::PM;
# ABSTRACT: Just a file to be indexed

our $VERSION = \'1.0\';
',
    'Dist-Metadata-Test-SubDir-1.5/README' => 'This "dist" is for testing Dist::Metadata.
'
  }
};


Makefile.PL

use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Information about a perl module distribution",
  "AUTHOR" => "Randy Stauner <rwstauner\@cpan.org>",
  "BUILD_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Spec::Functions" => 0,
    "Test::Fatal" => 0,
    "Test::MockObject" => "1.09",
    "Test::More" => "0.96"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Metadata",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Metadata",
  "PREREQ_PM" => {
    "Archive::Tar" => 1,
    "Archive::Zip" => "1.30",
    "CPAN::DistnameInfo" => "0.12",
    "CPAN::Meta" => "2.1",
    "Carp" => 0,
    "Digest" => "1.03",
    "Digest::MD5" => 2,
    "Digest::SHA" => 5,
    "File::Basename" => 0,
    "File::Find" => 0,
    "File::Spec::Native" => "1.002",
    "File::Temp" => "0.19",
    "List::Util" => 0,
    "Module::Metadata" => 0,
    "Path::Class" => "0.24",
    "Try::Tiny" => "0.09",
    "constant" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.925",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



